lib/src/cli/cli.dart

```diff
@@ -162,8 +162,31 @@ const _ignoredDirectories = {
 };
 
 bool _isPubspec(FileSystemEntity entity) {
-  final segments = p.split(entity.path).toSet();
-  if (segments.intersection(_ignoredDirectories).isNotEmpty) return false;
   if (entity is! File) return false;
   return p.basename(entity.path) == 'pubspec.yaml';
 }
+
+extension on Set<String> {
+  bool excludes(FileSystemEntity entity) {
+    final segments = p.split(entity.path).toSet();
+    if (segments.intersection(_ignoredDirectories).isNotEmpty) return true;
+    if (segments.intersection(this).isNotEmpty) return true;
+
+    for (var value in this) {
+      if (value.contains('/') && !value.contains('*')) {
+        if (value.endsWith('/')) {
+          value = '$value**';
+        } else {
+          value = '$value/**';
+        }
+      }
+
+      final glob = Glob(value);
+      if (glob.matches(entity.path)) {
+        return true;
+      }
```

Comment:
Why are we turning the `value` into a glob when there is a `/` but no `*` in the string? Could you elaborate on what this is doing exactly?

Reply:
For some reason if the `value` is  `dev/projects/dev` (example), `Glob(value)` wouldn't find a match. My workaround is to add the glob pattern based on the `value` so that `dev/projects/dev` and `dev/projects/dev/` would both convert to `dev/projects/dev/**` so that the glob can find a match!

Reply:
Hmm but if that's the case I think we can get away by doing it like this:

```suggestion
      if (value.contains('*') {
         return Glob(value).matches(entity.path);
      }
      return entity.path.startsWith(value);
```

Reply:
Tricky part is that `entity.path.startsWith(value)` would always return false since the ignored package (value), isn't the full path like `Users/user/my_project/packages/dev/...`. 

So if I ran `very_good packages get --recursive --ignore packages/dev`, this would still get packages in `packages/dev`

Technically using `entity.path.contains(value)` will work, but not sure if this would be ideal.

wdyt?

Reply:
Just synced with @willhlas on this and the use case of a string like `dev/projects/dev` without a `*` is not a valid glob according to the specs. So if we are only supporting the glob pattern we should not introduce non-glob patterns for matching purposes.

So let's revert to the original implementation of only doing a `return Glob(value).matches(entity.path)`